File: ng2-ts/src/app/component/sign-in.component.ts
import { Component, OnInit } from '@angular/core';
import { Router } from '@angular/router';

import { User } from '../model/user';
import { AuthService } from '../service/auth.service';

@Component({
    selector: 'sign-in',
    templateUrl: './sign-in.component.html'
})
export class SignInComponent { 

    model: User;
    errorMessage: string;
    loading: boolean;

    constructor(private router: Router, private authService: AuthService) {   
        this.model = new User();
	 }

    signIn() {
      this.loading = true;
      this.errorMessage = null;
      setTimeout(() => { 
              this.authService.signIn(this.model)
                  .subscribe((result:User) => {
                    this.loading = false;
                     if (result) {
                       this.router.navigate(['']);
                     }
            },
            (err: any) => {
              this.loading = false;
              if (err.status == 401) {
                this.errorMessage = "Invalid username or password";
              }  
            });
      }, 1000);
    }

    signUp() {
        this.router.navigate(['sign-up']);
    }

}

File: ng2-dart/lib/component/sign-in.component.dart
import 'package:angular2/core.dart';
import 'package:angular2/router.dart';

import '../model/user.dart';

import '../service/auth.service.dart';

@Component(
	selector: 'sign-in',
	templateUrl: './sign-in.component.html')
class SignInComponent {
	
	User model;
	
	String errorMessage;
	bool loading;

	final Router _router;
	final AuthService _authService;

	SignInComponent(this._router, this._authService);

	signIn() {
		loading = true;
		errorMessage = null;

		this._authService.signIn(model)
			.then((User user) {
				this.loading = false;
				this._router.navigate(['']);
			})
			.catchError((error) {
				this.loading = false;
				this.errorMessage = "Nope";
			});
	}

	signUp() {
		_router.navigate(['sign-up']);
	}
}
